lib/base/json.cpp

```diff
@@ -94,7 +94,7 @@ void EncodeNamespace(JsonEncoder<prettyPrint>& stateMachine, const Namespace::Pt
 
 	ObjectLock olock(ns);
 	for (const Namespace::Pair& kv : ns) {
-		stateMachine.Key(Utility::ValidateUTF8(kv.first));
+		stateMachine.Key(kv.first);
```

Comment:
How exactly these ones copy less than the to-be-closed #9519 ...

Reply:
With my changes it will only prevent being copied within the `ValidateUTF8()` method, though not what other methods do with the passed data, i.e. your changes copy the data 1x inside the validate utf8 method and is passed by value to `stateMachine.Key()`. With my changes the data is passed by value to `stateMachine.Key()` and validated without copying.

Reply:
## IMAO

### In my PR

1. Utility::ValidateUTF8 gets kv.first by ref,
2. **copies** it while sanitising and
3. returns the copy by move.
4. stateMachine.Key move-constructs Utility::ValidateUTF8's result and
5. moves it...

### In your PR

1. stateMachine.Key **copy-constructs** from kv.first,
2. Utility::ValidateUTF8-s the copy in-place and
3. moves it...

### Summary

Our PRs are equal at this point, but mine doesn’t require refactoring.